modules/ipc/include/hephaestus/ipc/zenoh/action_server.h

```diff
@@ -0,0 +1,254 @@
+//================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <future>
+#include <optional>
+
+#include <magic_enum.hpp>
+
+#include "hephaestus/concurrency/message_queue_consumer.h"
+#include "hephaestus/ipc/common.h"
+#include "hephaestus/ipc/publisher.h"
+#include "hephaestus/ipc/zenoh/internal/action_server_client_helper.h"
+#include "hephaestus/ipc/zenoh/service.h"
+#include "hephaestus/ipc/zenoh/session.h"
+#include "hephaestus/ipc/zenoh/types/action_server_types.h"
+#include "hephaestus/ipc/zenoh/types/action_server_types_protobuf.h"  // NOLINT(misc-include-cleaner)
+
+namespace heph::ipc::zenoh {
+
+enum class ActionServerTriggerStatus : uint8_t {
+  SUCCESSFUL = 0,
+  REJECTED = 1,
+};
```

Comment:
```suggestion
namspace action_server {
enum class TriggerStatus : uint8_t {
  SUCCESSFUL = 0,
  REJECTED = 1,
};
}
```
nit, style. If you want to avoid a long name which consists of scope + purpose, you could simply place it inside the class with public visibility, or use a namespace.

The proposed solution of ScopePurpose works as well and is the one which I use in Protobuf enums, which always have global visibility. Not my favorite style though.

Reply:
the problem of putting inside the class is that in this case the class is templated so to use it outside you need to repeat the template, very annoying

Reply:
namespace is great